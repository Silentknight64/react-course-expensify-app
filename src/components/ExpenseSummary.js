import React from 'react';
import numeral from 'numeral';
import getTotalExpenses from '../selectors/expenses-total';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';

export class ExpenseSummary extends React.Component {
	render() {
		return (
			<div>
				<h1>Viewing {this.props.expenses.length} {this.props.expenses.length === 1 ? 'expense' : 'expenses'} totalling {numeral(getTotalExpenses(this.props.expenses) / 100).format('$0,0.00')}</h1>
			</div>
		)	
	}
}

const mapStateToProps = (state) => ({
	expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseSummary);