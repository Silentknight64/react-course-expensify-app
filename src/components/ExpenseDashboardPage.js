import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

class ExpenseDashboardPage extends React.Component {
	render() {
		return (
			<div>
				<ExpenseSummary />
				<ExpenseListFilters />
				<ExpenseList />
			</div>		
		)	
	}
}

export default ExpenseDashboardPage;
