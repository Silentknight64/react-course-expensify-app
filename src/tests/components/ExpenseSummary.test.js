import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses, { noExpenses, oneExpense } from '../fixtures/expenses';

test('should render ExpenseSummary correctly with 0 expenses', () => {
	const wrapper = shallow(<ExpenseSummary expenses={noExpenses} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with 1 expense', () => {
	const wrapper = shallow(<ExpenseSummary expenses={oneExpense} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with 3 expenses', () => {
	const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
	expect(wrapper).toMatchSnapshot();
})