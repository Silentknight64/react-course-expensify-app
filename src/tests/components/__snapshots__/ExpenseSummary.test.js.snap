// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 0 expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with 3 expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </h1>
</div>
`;
